ios/engine/KMEI/KeymanEngineTests/KeymanPackageTests.swift

```diff
@@ -58,6 +58,98 @@ class KeymanPackageTests: XCTestCase {
+    let destinationDirectory = cacheDirectory.appendingPathComponent("doesnotexist")
+
+    do {
+      // clear directory
+      try KeymanPackage.clearDirectory(destination: destinationDirectory)
+      } catch {
+        XCTFail("error clearing the nonexistent directory \(error)")
+      }
+  }
+
+  func testKeyboardPackage_clearEmptyDirectory_throwsNoError() throws {
+    let cacheDirectory = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)[0]
+    let destinationDirectory = cacheDirectory.appendingPathComponent("destination")
+
+    do {
+      // create directory
+      try FileManager.default.createDirectory(
+        atPath: destinationDirectory.path,
+        withIntermediateDirectories: false,
+        attributes: nil
+      )
+
+      // clear directory
+      try KeymanPackage.clearDirectory(destination: destinationDirectory)
+      } catch {
+        XCTFail("error clearing the empty directory \(error)")
+      }
+    
+    let fileArray = try FileManager.default.contentsOfDirectory(atPath: destinationDirectory.path)
+    XCTAssert(fileArray.count == 0, "directory still contains /(fileArray.count) items")
+  }
+
+  func testKeyboardPackage_clearNonEmptyDirectory_directoryIsEmpty() throws {
+    let cacheDirectory = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)[0]
+    let destinationDirectory = cacheDirectory.appendingPathComponent("destination")
+
+    do {
+      // create directory
+      try FileManager.default.createDirectory(
+        atPath: destinationDirectory.path,
+        withIntermediateDirectories: false,
+        attributes: nil
+      )
+
+      // add some files
+      FileManager.default.createFile(atPath: destinationDirectory.appendingPathComponent("fileone").path, contents: nil)
+      FileManager.default.createFile(atPath: destinationDirectory.appendingPathComponent("filetwo").path, contents: nil)
+      FileManager.default.createFile(atPath: destinationDirectory.appendingPathComponent("filethree").path, contents: nil)
+
+      // clear directory
+      try KeymanPackage.clearDirectory(destination: destinationDirectory)
+      } catch {
+        XCTFail("error clearing the empty directory \(error)")
+      }
+    
+    let fileArray = try FileManager.default.contentsOfDirectory(atPath: destinationDirectory.path)
+    XCTAssert(fileArray.count == 0, "directory still contains /(fileArray.count) items")
+  }
+
+  func testKeyboardPackage_extractTwice_noDuplicateFileError() throws {
+    let cacheDirectory = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)[0]
+    let khmerPackageZip = cacheDirectory.appendingPathComponent("khmer_angkor.kmp.zip")
+    try ResourceFileManager.shared.copyWithOverwrite(from: TestUtils.Keyboards.khmerAngkorKMP, to: khmerPackageZip)
+
+    let destinationFolderURL = cacheDirectory.appendingPathComponent("khmer_angkor")
+
+    // Requires that the source file is already .zip, not .kmp.  It's a ZipUtils limitation.
```

Comment:
Is this restriction still true with the zip framework change? (If not, then that's a follow-up PR on alpha that we could pop in)

Reply:
I omitted '.zip' from the archive name in the unit tests, and they all passed ok -- so the new framework does not have that limitation.  I'll change in the app as a separate PR.